File: app/src/main/java/com/android/virgilsecurity/virgilback4app/SplashActivity.java
package com.android.virgilsecurity.virgilback4app;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;

import com.android.virgilsecurity.virgilback4app.auth.SignInControlActivity;
import com.android.virgilsecurity.virgilback4app.chat.contactsList.ThreadsListActivity;
import com.parse.ParseUser;

public class SplashActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        if (ParseUser.getCurrentUser() != null) {
            ThreadsListActivity.startWithFinish(this);
        } else {
            SignInControlActivity.startWithFinish(this);
        }
    }

    @Override public void onBackPressed() {

    }
}

File: app/src/main/java/com/android/virgilsecurity/virgilback4app/SplashActivity.kt
package com.android.virgilsecurity.virgilback4app

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity

import com.android.virgilsecurity.virgilback4app.auth.SignInControlActivity
import com.android.virgilsecurity.virgilback4app.chat.contactsList.ThreadsListActivity
import com.parse.ParseUser

class SplashActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        if (ParseUser.getCurrentUser() != null) {
            ThreadsListActivity.startWithFinish(this)
        } else {
            SignInControlActivity.startWithFinish(this)
        }
    }

    override fun onBackPressed() {
        // Must be empty, so we can't press back from the splash screen
    }
}
